### PR TITLE
Set docker image in valohai.prepare

### PR DESCRIPTION
**Pull request by DrazenDodik**
Allow setting a custom docker image when running valohai.prepare

### CODE REVIEW

File: valohai/internals/merge.py [hunk 1]
```diff
@@ -74,4 +74,8 @@ def _merge_step(original: Step, parsed: Step) -> Step:
         skip_missing_b=True,
     )
 
+    if(parsed.image is not None) :
```
Comment (JuhaKiili) [inline valohai/internals/merge.py:77]: Parentheses are not needed.

`if parsed.image is not None:`

File: tests/test_yaml/test2.py [hunk 2]
```diff
@@ -16,4 +16,4 @@ def prepare(a, b):
     print(f"this is fake method {a} {b}")
 
 
-valohai.prepare(step="foobar1", default_parameters=params, default_inputs=inputs)
+valohai.prepare(step="foobar1", default_parameters=params, default_inputs=inputs, image="hope/thiswillnotchange:1234")
```
Comment (JuhaKiili) [inline tests/test_yaml/test2.py:19]: I would like to see this test pass without adding the image to test2.py.

If user has an original `valohai.yaml` with `image: hope/thiswillnotchange:1234` and then doesn't have `image` set in their `prepare()` call, then we should let the original `image: hope/thiswillnotchange:1234` be untouched.

As a rule of thumb: Anything that is **not** defined in `.py` file should stay as it is when merging.

This test2 should test for that behavior.

**Review by JuhaKiili — CHANGES_REQUESTED** [review]

File: valohai/internals/merge.py (hunk 1 above)
Comment (DrazenDodik) [reply valohai/internals/merge.py]: Fixed ✅

File: tests/test_yaml/test2.py (hunk 2 above)
Comment (DrazenDodik) [reply tests/test_yaml/test2.py]: Changed 🥷 ✅

File: valohai/internals/parsing.py [hunk 3]
```diff
@@ -88,11 +90,15 @@ def process_valohai_prepare_call(self, node):
...
 
 
 def parse(source):
     tree = ast.parse(source)
     parser = PrepareParser()
     parser.visit(tree)
-    result = namedtuple("result", ["step", "parameters", "inputs"])
-    return result(step=parser.step, parameters=parser.parameters, inputs=parser.inputs)
+    result = namedtuple("result", ["step", "parameters", "inputs", "image"])
+    print(result)
```
Comment (JuhaKiili) [inline valohai/internals/parsing.py:103]:
```suggestion
```

File: valohai/utils.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ def prepare(*, step: str, default_parameters: dict = {}, default_inputs: dict =
```
Comment (JuhaKiili) [inline valohai/utils.py:23]: Add :param like
```
:param image: Default docker image
```

**Review by JuhaKiili — CHANGES_REQUESTED** [review]
Some final polish

**Review by JuhaKiili — APPROVED** [review]

**Pull request merged**